src/AvaloniaEdit.TextMate/TextMate.cs

```diff
@@ -61,17 +70,93 @@ public void SetGrammar(string scopeName)
+                        {
+                            Application.Current.RequestedThemeVariant = ThemeVariant.Light;
+                        }
+                        else
+                        {
+                            Application.Current.RequestedThemeVariant = ThemeVariant.Dark;
+                        }
+                    
+                        ApplyBrushAction("editor.background",brush =>window.Background = brush);
+                        ApplyBrushAction("editor.foreground",brush =>window.Foreground = brush);
+                    }
+
+                    ApplyBrushAction("editor.background",brush =>_editor.Background = brush);
+                    ApplyBrushAction("editor.foreground",brush =>_editor.Foreground = brush);
+                    
+                    if (_defaultSelectionBrush == null)
+                    {
+                        _defaultSelectionBrush = _editor.TextArea.SelectionBrush;
+                    }
+                    
+                    if (!ApplyBrushAction("editor.selectionBackground",
+                            brush => _editor.TextArea.SelectionBrush = brush))
+                    {
+                        _editor.TextArea.SelectionBrush = _defaultSelectionBrush;
+                    }
+
+                    if (!ApplyBrushAction("editor.lineHighlightBackground",
+                            brush =>
+                            {
+                                _editor.TextArea.TextView.CurrentLineBackground = brush;
+                                _editor.TextArea.TextView.CurrentLineBorder = new Pen(brush); // Todo: VS Code didn't seem to have a border but it might be nice to have that option. For now just make it the same..
+                            }))
+                    {
+                        _editor.TextArea.TextView.CurrentLineBackground = new SolidColorBrush(CurrentHighlightRendererDefaultBackground);
+                        _editor.TextArea.TextView.CurrentLineBorder = new Pen(new SolidColorBrush(CurrentHighlightRendererDefaultBorder));
+                    }
+
+                    
+                    //Todo: looks like the margin doesn't have a active line highlight, would be a nice addition
+                    if (!ApplyBrushAction("editorLineNumber.foreground",
+                            brush => _editor.LineNumbersForeground = brush))
+                    {
+                        _editor.LineNumbersForeground = _editor.Foreground;
+                    }
+                    
+                    AppliedTheme?.Invoke(this,this);
+                }
             }
+            
+            //These come out of CurrentHighlightRenderer sealed class. 
+            public static readonly Color CurrentHighlightRendererDefaultBackground = Color.FromArgb(22, 20, 220, 224);
+            public static readonly Color CurrentHighlightRendererDefaultBorder = Color.FromArgb(52, 0, 255, 110);
```

Comment:
These hardcoded colors should be avoided and removed in CurrentLineHighlightRenderer class as well. Colors should be defined by XAML themes. But that's for another day, I guess.

Reply:
I at least got rid of the Duplicate definition,  but I agree, this should be some property up higher on the textview, not in the background highlight renderer. Just not today